### PR TITLE
[FIX] mail: avoid notification if the channel is not a chat.

### PR DESCRIPTION
**Pull request by phenix-factory**
task-2387822

### CODE REVIEW

File: addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js [hunk 1]
```diff
@@ -231,6 +231,10 @@ function factory(dependencies) {
                 return;
             }
 
+            if (!channel.isChatChannel) {
+                return;
+            }
```
Comment (seb-odoo) [inline addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js:236]: This is returning too early. The update of the counter (last line of method) must still be executed in this case.

Should just prevent `_notifyNewChannelMessageWhileOutOfFocus`, similar to condition with `isOdooFocused` below.

**Review by seb-odoo — CHANGES_REQUESTED** [review]
See comment below, also typo in commit title is->if (and no need for dot at the end)

File: addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js (hunk 1 above)
Comment (phenix-factory) [reply addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js:236]: I don't understand the role of the focus here.
We don't want notification from a channel at all. Maybe this isn't the best way to resolve this issue.

Comment (seb-odoo) [reply addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js:236]: If the Odoo window is currently focused, we don't want to send a native notification. Those are only useful if the user is looking elsewhere, to bring attention to Odoo.

But if Odoo is already "focused", we just want to send a "normal" notification inside Odoo itself (with `notify` or similar).

You don't really need to worry about any of this for this fix though, you should just ensure the method is not called for channels, and my only suggestion is to move your condition to strictly restrict this method call, and not the code that come afterwards in the handler.

File: addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js [hunk 2]
```diff
@@ -231,6 +232,13 @@ function factory(dependencies) {
                 return;
             }
 
+            if (!channel.isChatChannel && !isOdooFocused) {
+                this._notifyNewChannelMessageWhileOutOfFocus({
+                    channel,
+                    message,
+                });
+            }
+
```
Comment (seb-odoo) [inline addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js:241]: We don't want to call this method another time. We just need to make sure `!channel.isChatChannel` when calling it below. We still don't want to call it at all in the case `isChatWithOdooBot` for example.

Comment (phenix-factory) [reply addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js]: It was bugged anyway, got 2 notification.
Look like it's working now

**Review by alexkuhn — APPROVED** [review]
LGTM, just remove task-2388923 in commit message, because this task has been canceled.
Also side-note if it ever happen in future: when a commit relates to several tasks, have them listed by increasing order of ids.

**Comment by alexkuhn** [comment]
Oh and also runbot is red due to failing test. I'm suspecting the test need to clarify channel in use must be a chat channel (e.g. a dm chat).

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**